scripts/get_module_version.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# read from stdin if no argument is supplied
+if [ $# -eq 0 ]; then
+    read -r VERSION < /dev/stdin
+else
+    VERSION=$1
+fi
+
+# drop leading v if present
+if [[ "$VERSION" = v* ]]; then
+    VERSION="${VERSION:1}"
+fi
+
+MAJOR=$(echo "$VERSION" | cut -d. -f1)
+MINOR=$(echo "$VERSION" | cut -d. -f2)
+REST=$(echo "$VERSION" | cut -d. -f3-)
+
+if [[ $MAJOR == "2" ]]; then
+    echo "0.$MINOR.$REST"
+fi
+
+# starting with Prometheus 3.0 the go module will start at 0.300
+if [[ $MAJOR == "3" ]]; then
+    printf "0.3%02d.$REST\n" "$MINOR"
+fi
```

Comment:
Thanks, I wasn't sure about whether this would work on macos.

How about we keep the option to pass the version as a string arg and read `VERSION` if no argument is supplied. Passing `${{ github.ref_name }}` as a string in an action is less awkward I think.